File: hrms_admin/designationmaster/designationmasterservice/designationmasterservice.py
from hrms_admin.models import designationmaster


class dmdesignation:
    @classmethod
    def saveDesignation(self, designationName,created_by,created_date):
        try:
            saveqry = designationmaster(designation_name=designationName,created_by=created_by,created_date=created_date)
            saveqry.save()
            saveqrysuccessobj = {
                'response': "success"
            }
            return saveqrysuccessobj
        except Exception, err:
            saveqryfailureobj = {
                'response': "Failure"
            }
            return {'failureobj': saveqryfailureobj}


    @classmethod
    def listDesignation(self):
        try:
            listqry = designationmaster.objects.all()
            designationList = []

            for values in listqry:
                designationList.append({
                    'des_id' : values.designation_id,
                    'des_name' : values.designation_name,
                })
            return designationList

        except Exception, err:
            saveqryfailureobj = {
                'response': "Failure"
            }
            return saveqryfailureobj

    @classmethod
    def deleteDesignation(self, id):
        try:
            delqry = designationmaster.objects.get(designation_id=id)
            delqry.delete()
            successobj = {
                'successmsg': "success",
            }
            return successobj

        except Exception, err:
            saveqryfailureobj = {
                'response': "Failure"
            }
            return saveqryfailureobj

    @classmethod
    def geteditDesignationdata(cls,id):
        try:
            getqry = designationmaster.objects.get(designation_id=id)
            dataobj = {
                'des_id': getqry.designation_id,
                'des_name': getqry.designation_name,
            }
            return dataobj
        except Exception, err:
            saveqryfailureobj = {
                'response': "Failure"
            }
            return saveqryfailureobj

    @classmethod
    def updateDesignation(self, id, designationName,updated_by,updated_date):
        try:
            getqry = designationmaster.objects.get(designation_id=id)

            getqry.designation_name = designationName
            getqry.updated_date=updated_date
            getqry.updated_by=updated_by

            getqry.save()

            saveqrysuccessobj = {
                'response': "success"
            }
            return saveqrysuccessobj

        except Exception, err:
            saveqryfailureobj = {
                'response': "Failure"
            }
            return {'failureobj': saveqryfailureobj}

File: static/components/admin/components/designationmaster/designationmasterService/designationmasterService.js
angular.module('adminApp.designationService',[])

.service('designationService',['$http', function($http){

    var designation ={};

    designation.addDesignation = function(designationname,success, failure){

        $http.post('api/savedesignation/',{
            'des_name' : designationname,

        }).then(success, failure)
    }

    designation.listDesignation = function(success, failure){
            $http.get('api/listdesignation/').then(success, failure)
    }

    designation.deleteDesignation = function(des_id, success, failure){
         $http.delete('api/deletedesignation/?id='+des_id).then(success, failure)
    }

    designation.geteditdata = function(id, success, failure){
        $http.get('api/geteditdesignationdata/?id='+id).then(success, failure)
    }

    designation.updateDesignation = function(desData, success, failure){
        $http.post('api/updatedesignation/',{
            'id': desData.desid,
            'des_name' : desData.desName,
        }).then(success, failure)
    }

    return designation;

}])

